services-custom/s3-event-notifications/README.md

```diff
@@ -0,0 +1,36 @@
+# S3 Event Notifications
+
+## Overview
+
+This module contains the classes used to represent Amazon S3 Event Notifications.
+
+
+## Deserialization
+
+To convert a json notification to java classes, use the static methods 
+available on `S3EventNotification`:
+
+```java
+String json = "..."; // the notification as json
+S3EventNotification event = S3EventNotification.fromJson(json);
+event.getRecords().forEach(rec -> println(rec.toString()));
+```
+
+Any missing fields of the json will be null in the resulting object. 
+Any extra fields will be ignored.
+
+
+## Serialization
+
+To convert an instance of `S3EventNotification` to json, use the `.toJson()`
+or `toJsonPretty()` method:
+
+```java
+S3EventNotification event = new S3EventNotification(...);
+String json = event.toJson();
+String jsonPretty = event.toJsonPretty();
+```
+
+`GlacierEventData`, `ReplicationEventData`, `IntelligentTieringEventData` and `LifecycleEventData`
+will be excluded from the json if null. Any other null fields of the object will be 
+serialized in the json as `null`.
```

Comment:
Why the difference?

Reply:
These fields are only present in the event json under specific circumstances. From the **[Event Structure](https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html)** page:

> The glacierEventData key is only visible for s3:ObjectRestore:Completed events.
> The replicationEventData key is only visible for replication events.
> The intelligentTieringEventData key is only visible for S3 Intelligent-Tiering events.
> The lifecycleEventData key is only visible for S3 Lifecycle transition events.

Reply:
Do we need the other null fields to be explicitly marshalled as null, or could we exclude them as well?

Reply:
The other fields are marshalled as null in v1, so I think we should keep the same behaviour